Altinn.Platform.Authentication.SystemIntegrationTests/Utils/MaskinPortenTokenGenerator.cs

```diff
@@ -0,0 +1,143 @@
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Text.Json;
+using Altinn.Platform.Authentication.SystemIntegrationTests.Domain;
+using Microsoft.IdentityModel.Tokens;
+using Xunit;
+using JwtRegisteredClaimNames = Microsoft.IdentityModel.JsonWebTokens.JwtRegisteredClaimNames;
+
+namespace Altinn.Platform.Authentication.SystemIntegrationTests.Utils;
+
+/// <summary>
+/// Helper class to generate a machineporten token
+/// </summary>
+public class MaskinPortenTokenGenerator
+{
+    private static string ToStandardBase64(string? base64Url)
+    {
+        Assert.True(base64Url != null, "Base64 url should not be null");
+        var base64 = base64Url.Replace('-', '+').Replace('_', '/');
+        switch (base64.Length % 4)
+        {
+            case 2:
+                base64 += "==";
+                break;
+            case 3:
+                base64 += "=";
+                break;
+        }
+
+        return base64;
+    }
+
+    public static Task<string> GenerateJwt(Jwk jwk, EnvironmentHelper.MaskinportenClient client)
```

Comment:
Brukes dette i test på noen måte

Reply:
Ja, bruker den for å hente Maskinportentoken:

 using var client = new HttpClient();
        var requestContent = new FormUrlEncodedContent([
            new KeyValuePair<string, string>("grant_type", "urn:ietf:params:oauth:grant-type:jwt-bearer"),
            new KeyValuePair<string, string>("assertion", jwt)
        ]);

        var response = await client.PostAsync("https://test.maskinporten.no/token", requestContent);
        
        
        eller var det noe spesielt du tenkte på der, @TheTechArch ?